ci/gpu/build.sh

```diff
@@ -108,11 +108,18 @@ conda config --show-sources
 conda list --show-channel-urls
 
 function install_dask {
-    # Install the main version of dask, distributed, and streamz
-    gpuci_logger "Install the main version of dask, distributed, and streamz"
+    # Install the conda-forge or nightly version of dask and distributed
+    gpuci_logger "Install the conda-forge or nightly version of dask and distributed"
     set -x
-    pip install "git+https://github.com/dask/distributed.git@$DASK_DISTRIBUTED_GIT_TAG" --upgrade --no-deps
-    pip install "git+https://github.com/dask/dask.git@$DASK_DISTRIBUTED_GIT_TAG" --upgrade --no-deps
+    if [[ "${INSTALL_DASK_MAIN}" == 1 ]]; then
+        gpuci_logger "gpuci_mamba_retry install dask/label/dev::dask --force-reinstall"
+        gpuci_mamba_retry install dask/label/dev::dask --force-reinstall
+    else
+        gpuci_logger "gpuci_mamba_retry install -c conda-forge dask"
+        gpuci_mamba_retry install -c conda-forge dask
+    fi
```

Comment:
Do we need to specify the channels in either of these? Hoping having the channels according to PR ( https://github.com/rapidsai/cudf/pull/10182 ) already handles this for us

Or if that doesn't work, what kinds of issues do we see?

Reply:
> Or if that doesn't work, what kinds of issues do we see?

Here the job url where the force reinstall wasn't successfully installing `dask` from `dask/label/dev` and it was using existing `conda-forge`: https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci/job/cudf/job/prb/job/cudf-gpu-test/CUDA=11.2,GPU_LABEL=driver-495,LINUX_VER=ubuntu18.04,PYTHON=3.9/6819/consoleText

```bash
++ gpuci_logger 'gpuci_mamba_retry install -c dask/label/dev dask --force-reinstall'

[32mgpuCI logger[0m » [02/15/2022 00:33:45]
[32m┌──────────────────────────────────────────────────────────────────────────┐[0m
[32m|    gpuci_mamba_retry install -c dask/label/dev dask --force-reinstall    |[0m
[32m└──────────────────────────────────────────────────────────────────────────┘[0m

++ gpuci_mamba_retry install -c dask/label/dev dask --force-reinstall
conda-forge/linux-64     Using cache
conda-forge/noarch       Using cache
gpuci/noarch              
dask/label/dev/noarch     
nvidia/noarch             
gpuci/linux-64            
rapidsai-nightly/linux-6  
pytorch/linux-64          
dask/label/dev/linux-64   
rapidsai-nightly/noarch   
rapidsai/linux-64         
rapidsai/noarch           
pytorch/noarch            
nvidia/linux-64           
Transaction

  Prefix: /opt/conda/envs/rapids

  Updating specs:

   - dask


  Package   Version  Build         Channel                Size
────────────────────────────────────────────────────────────────
  Change:
────────────────────────────────────────────────────────────────

  - dask   2022.1.1  pyhd8ed1ab_0  installed                  
  + dask   2022.1.1  pyhd8ed1ab_0  conda-forge/noarch     5 KB

  Summary:

  Change: 1 packages

  Total download: 5 KB

────────────────────────────────────────────────────────────────

Confirm changes: [Y/n] 
                  __    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
███████████████/  /██/  /██/  /██/  /████████████████████████
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗
        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗
        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║
        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║
        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║
        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝

        mamba (0.15.2) supported by @QuantStack

        GitHub:  https://github.com/mamba-org/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████


Looking for: ['dask']


Pinned packages:
  - python 3.9.10


Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
Finished dask                                 (00m:00s)               5 KB     18 KB/s
```

Reply:
Think I have an of why we're seeing these weird environment solves - it turns out that nightlies are seen as downgrades from the stable package 😕 running a `conda search`:

```
$ mamba search --override-channels -c dask/label/dev -c conda-forge dask
...
dask                 2022.02.0a220214 py38_g820427a3_1  dask/label/dev
dask                 2022.02.0a220214  py_gd3f3012d_2  dask/label/dev
dask                 2022.02.0a220215  py_g00040a54_7  dask/label/dev
dask                 2022.02.0a220215  py_g0efad518_6  dask/label/dev
dask                 2022.02.0a220215  py_g973b09c9_5  dask/label/dev
dask                 2022.02.0a220215  py_gf25f1fc1_8  dask/label/dev
dask                        2022.2.0    pyhd8ed1ab_0  conda-forge
```

Looks like this is because dask/distributed zero-pads their release tags but not the releases themselves - I can look into handling this in the nightly recipes (i.e. remove the zero), but wondering if there's a reason this is done @jakirkham?

Reply:
With https://github.com/dask/dask/pull/8728 and https://github.com/dask/distributed/pull/5816 in, we are now producing Dask nightlies with a patch version higher than the latest stable release, so running something like

```
gpuci_mamba_retry update dask
```

Should just pick up the latest Dask nightlies now.

> Do we need to specify the channels in either of these?

I think we might still want to specify the `-c conda-forge` for the case where `INSTALL_DASK_MAIN=0`, as I imagine there we want to force a downgrade to the stable Dask packages?

Reply:
@jakirkham Yup, that seemed to have worked now:
```
[32mgpuCI logger[0m » [02/23/2022 16:14:51]
[32m┌──────────────────────────────────────────────────────────────────────────┐[0m
[32m|    Install the conda-forge or nightly version of dask and distributed    |[0m
[32m└──────────────────────────────────────────────────────────────────────────┘[0m

++ set -x
++ [[ 1 == 1 ]]
++ gpuci_logger 'gpuci_mamba_retry update dask'

[32mgpuCI logger[0m » [02/23/2022 16:14:51]
[32m┌─────────────────────────────────────┐[0m
[32m|    gpuci_mamba_retry update dask    |[0m
[32m└─────────────────────────────────────┘[0m

++ gpuci_mamba_retry update dask
conda-forge/linux-64     Using cache
conda-forge/noarch       Using cache
gpuci/noarch              
dask/label/dev/linux-64   
gpuci/linux-64            
pytorch/linux-64          
rapidsai-nightly/noarch   
rapidsai-nightly/linux-6  
nvidia/linux-64           
rapidsai/noarch           
nvidia/noarch             
rapidsai/linux-64         
dask/label/dev/noarch     
pytorch/noarch            
Transaction

  Prefix: /opt/conda/envs/rapids

  Updating specs:

   - dask
   - ca-certificates
   - certifi
   - openssl


  Package                 Version  Build              Channel                     Size
────────────────────────────────────────────────────────────────────────────────────────
  Upgrade:
────────────────────────────────────────────────────────────────────────────────────────

  - dask         2022.02.1a220222  py_g9a266a04_29    installed                       
  + dask         2022.02.1a220223  py_g8a99ac7c_31    dask/label/dev/noarch       5 KB
  - distributed  2022.02.1a220222  py39_g9a266a04_29  installed                       
  + distributed  2022.02.1a220223  py39_g8a99ac7c_31  dask/label/dev/linux-64     1 MB

  Summary:

  Upgrade: 2 packages

  Total download: 1 MB

────────────────────────────────────────────────────────────────────────────────────────

Confirm changes: [Y/n] 
                  __    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
███████████████/  /██/  /██/  /██/  /████████████████████████
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗
        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗
        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║
        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║
        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║
        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝

        mamba (0.15.2) supported by @QuantStack

        GitHub:  https://github.com/mamba-org/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████


Looking for: ['dask']

```

Reply:
This should be ready for review & merge now.